### PR TITLE
Refactor to point, add agent factory, refactor agent tests to use factory

### PR DESCRIPTION
**Pull request by jrhackett**

### CODE REVIEW

File: server/src/point/point.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        return self.y
+
+    def set_point(self, x, y):
+        self.x = x
+        self.y = y
+
+    def set_x(self, x):
+        self.x = x
+
+    def set_y(self, y):
```
Comment (dombozzuto) [inline server/src/point/point.py]: my only criticism would be that getters and setters are not really Pythonic (as basically all class variables are already public).  point could probably be as simple as:
```
class Point:
 +    x: int = 0
 +    y: int = 0
 +
 +    def __init__(self, x=0, y=0):
 +        self.x = x
 +        self.y = y
+    def set_point(self, x, y):
 +        self.x = x
 +        self.y = y
```

File: server/src/point/point.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+
+class Point:
+    x: int = 0
+    y: int = 0
+
+    def __init__(self, x=0, y=0):
```
Comment (dombozzuto) [inline server/src/point/point.py]: minor, but you could type hint the defaults too (helps pycharm tell if you if you accidently set an int to a float, for example):
```
def __init__(self, x: int = 0, y: int = 0):
```

File: server/src/agent/agent_factory.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+class AgentFactory:
+
+    @staticmethod
+    def make_agent(agent_type: AgentType):
+        if agent_type == AgentType.SCOUT:
+            return impl.Scout()
+        if agent_type == AgentType.WARRIOR:
+            return impl.Warrior()
+        if agent_type == AgentType.WORKER:
+            return impl.Worker()
```
Comment (dombozzuto) [inline server/src/agent/agent_factory.py]: this is a beautiful factory

File: server/src/point/point.py (hunk 1 above)
Comment (jrhackett) [reply server/src/point/point.py]: That's fair. I'm not a fan of all class variables being public but I suppose we should follow convention.

**Pull request merged**